samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -103,8 +189,38 @@ static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 	}
 }
 
-static void stream_recv_cb(struct bt_bap_stream *stream,
-			   const struct bt_iso_recv_info *info,
+#if defined(CONFIG_LIBLC3)
+static void stream_recv_lc3_codec(struct bt_bap_stream *stream, const struct bt_iso_recv_info *info,
+				  struct net_buf *buf)
+{
+	static uint32_t lc3_recv_cnt;
+	struct broadcast_sink_stream *sink_stream =
+		CONTAINER_OF(stream, struct broadcast_sink_stream, stream);
+
+	lc3_recv_cnt++;
+	if ((lc3_recv_cnt % 1000U) == 0U) {
+		printk("Received %u total ISO packets\n", lc3_recv_cnt);
+	}
+
+	if (lc3_decoder == NULL) {
+		printk("LC3 decoder not setup, cannot decode data.\n");
+		return;
+	}
+
+	if ((info->flags & BT_ISO_FLAGS_VALID) == 0) {
+		printk("Bad packet: 0x%02X\n", info->flags);
+		sink_stream->in_buf = NULL;
+		sink_stream->in_buf_len = 0;
```

Comment:
This implies that `sink_stream->in_buf` may be non-null here. 

If that is the case, then we need to `net_buf_unref` it first.